data/generalhw_scripts/flash_sd_rootless.sh

```diff
@@ -25,38 +25,25 @@ hdd_size=$4
 echo "* Switch SD card to flasher"
 usbsdmux /dev/usb-sd-mux/id-$device_serial host
 
-# Cleanup target folder
-mkdir -p "$destination_folder/"
-rm -f $destination_folder/*.{raw,xz,iso,qcow2}
-echo "** Previous image deleted"
-
-image_to_flash_extension="${image_to_flash##*.}"
-if [ "$image_to_flash_extension" == "qcow2" ] ; then
-	uncompressed_filename="$(basename $image_to_flash .qcow2).raw"
-else
-	uncompressed_filename=$(basename $image_to_flash .xz)
-fi
-uncompressed_full_path="$destination_folder/$uncompressed_filename"
-
-# Extract compressed image, if needed
-if [[ "$image_to_flash_extension" == "xz" ]]; then
-	xzcat "$image_to_flash" --threads=0 "$uncompressed_full_path"
-	echo "*** xz image uncompressed"
-	# No need to resize since the SD card size is fixed
-elif [[ "$image_to_flash_extension" == "qcow2" ]]; then
-	qemu-img convert "$image_to_flash" "$uncompressed_full_path"
-	echo "*** qcow2 image uncompressed"
-fi
-
 # Check /dev/sdX_device is not mounted to prevent unexpected overwritting
 set +e
-sdX_device=$(readlink -f /dev/disk/by-id/usb-LinuxAut_sdmux_HS-SD_MMC_${device_serial}-0:0)
+device_link="/dev/disk/by-id/usb-LinuxAut_sdmux_HS-SD_MMC_${device_serial}-0:0"
+sdX_device=$(readlink -f $device_link)
 output=$(mount | grep /dev/$sdX_device)
 set -e
 if [ "$output" == "" ]; then
 	# Copy to SD card
 	echo "** Copy to SD card"
-	dd if=$uncompressed_full_path of=/dev/disk/by-id/usb-LinuxAut_sdmux_HS-SD_MMC_${device_serial}-0:0 oflag=sync bs=8M
+	image_to_flash_extension="${image_to_flash##*.}"
+	if [ "$image_to_flash_extension" == "qcow2" ] ; then
+		qemu-img dd -f qcow2 -O raw if=$image_to_flash of=$device_link bs=8M
+	elif [ "$image_to_flash_extension" == "xz" ] ; then
+		xzcat --threads=0 $image_to_flash | dd of=$device_link oflag=sync bs=8M
+	elif [ "$image_to_flash_extension" == "gz" ] ; then
+		zcat $image_to_flash | dd of=$device_link oflag=sync bs=8M
+	else
+		cat $image_to_flash | dd of=$device_link oflag=sync bs=8M
```

Comment:
Is there an advantage/disadvantage over  `dd if=$image_to_flash of=$device_link oflag=sync bs=8M` ?